### ISSUE TITLE
[documentation] Gradient of nn.MLP

### ISSUE THREAD

**Issue by jloganolson** [patrick-kidger/equinox#213]
What should I be doing here? 
```python
mlp = nn.MLP(in_size=2,out_size=1,width_size=80,depth=3,key=rng)
point = jax.random.uniform(rng,(2,))
print(mlp(point))
jax.grad(mlp)(point) #error: Gradient only defined for scalar-output functions. Output had shape: (1,).
```

**Comment by jloganolson** [comment]
FYI this worked but certainly there's something better we could do?
```python
rng=jax.random.PRNGKey(128)
mlp = nn.MLP(in_size=2,out_size=1,width_size=80,depth=3,key=rng)

def mlp_hack(x):
  out = mlp(x)
  return jp.reshape(out,())

point = jax.random.uniform(rng,(3,2))
jax.vmap(jax.grad(mlp_hack))(point)
```
